modules/education/src/main/java/edu/stanford/spezi/modules/education/videos/EducationViewModel.kt

```diff
@@ -27,11 +29,11 @@ internal class EducationViewModel @Inject constructor(
     private fun loadVideoSections() {
         viewModelScope.launch {
             _uiState.value = UiState.Loading
-            val result = educationRepository.getVideoSections()
-            _uiState.value = if (result.isSuccess) {
-                UiState.Success(EducationUiState(videoSections = result.getOrNull() ?: emptyList()))
-            } else {
-                UiState.Error("Failed to load video sections")
+            educationRepository.getVideoSections().onFailure {
+                logger.e(it) { "Failed to load video sections" }
+                _uiState.value = UiState.Error("Failed to load video sections")
```

Comment:
also here, extracting the string to string resources and updating the error message type to a `StringResource` would be great